enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,610 @@
+
+***GCP Labels***
+- An OpenShift administrator requests to add required labels to all GCP resources
+  created by OpenShift by adding it in `platform.gcp.userLabels`
+- OpenShift installer validates the labels defined in `platform.gcp.userLabels` and
+  adds these labels to all resources created during installation and also updates
+  `.status.platformStatus.gcp.resourceLabels` of the `infrastructure.config.openshift.io`
+- In-cluster operators refers `.status.platformStatus.gcp.resourceLabels` of the
+  `infrastructure.config.openshift.io` to add labels to the resources created later.
+
+***GCP Tags***
+- An OpenShift administrator requests to add required tags to all GCP resources
+  created by OpenShift by adding it in `platform.gcp.userTags`
+- OpenShift installer validates the tags defined in `platform.gcp.userTags` and
+  adds these tags to all resources created during installation and also updates
+  `.status.platformStatus.gcp.resourceTags` of the `infrastructure.config.openshift.io`
+- In-cluster operators refers `.status.platformStatus.gcp.resourceTags` of the
+  `infrastructure.config.openshift.io` to add tags to the resources created later.
+
+#### Variation [optional]
+
+### API Extensions
+Enhancement requires below modifications to the mentioned CRDs
+- Add `userLabels` and `userTags` fields to `platform.gcp` of the 
+  `installconfigs.install.openshift.io`
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: installconfigs.install.openshift.io
+spec:
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        properties:
+          platform:
+            properties:
+              gcp:
+                properties:
+                  organizationID:
+                    description: OrganizationID is the the organization identifier
+                      that will be used for deriving the tag key's namespaced name.
+                    type: string
+                  userLabels:
+                    additionalProperties:
+                      type: string
+                    description: UserLabels has additional keys and values that the installer
+                      will add as labels to all resources that it creates on GCP. Resources
+                      created by the cluster itself may not include these labels.
+                    type: object
+                  userTags:
+                    additionalProperties:
+                      type: string
+                    description: UserTags has additional keys and values that the installer
+                      will add as tags to all resources that it creates on GCP. Resources
+                      created by the cluster itself may not include these tags. Tag key
+                      and tag value should be the shortnames of the tag key and tag value
+                      resource.
+                    type: object
+```
+
+- Add `resourceLabels` and `resourceTags` fields to `status.platformStatus.gcp`
```

Comment:
Can you swap this schema for the Go definition, it's much easier to read and review than the schema

Reply:
comment incorporated.